### DISCUSSION TITLE
Store custom final object instances in object graph

### DISCUSSION THREAD

**Discussion by axkr** [microstream-one/microstream#232] [Q&A]
Is there an example how to store and retrieve "static final objects" (which are instantiated on startup) included in the objectgraph (for example as elements in a list)?
I'm thinkiing of a constant ID which is stored and mapped back after reading it into the "static final object"?

**Comment by hg-ms** [comment]
Currently there is no Example addressing static final objects explicitly.
What we currently do is:
1.	Static final fields are not persisted. They are part of the class, not of individual object instances. 
2.	If an object that references a static final object gets stored:

> -	The static final object gets persisted (if not already done), during that store it gets an ID assigned
> -	The referencing object gets persisted; it will store the ID of the static final object 

When loading, the persisted state is restored! Not the one that’s comes from the class definition.
This way we persist some kind of “copy” of the static final object and it is possible to have different instances of those in a storage.

There is also the possibility to register static final object as constants using a PersistenceRootResolver to enforce a stricter handling. But that’s currently not documented.

**Reply by axkr** [reply]
Ok, so no out-of.-the-box solution I guess.

Some background: 

I have these symbols which occur very often in the graph, which should be seen as "constant symbols" which are only initialized at startup:
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/expression/S.java

The goal is to store parsed package source codes (contexts) as persistent object graph for a faster loading on next startup.

**Comment by hg-ms** [comment]
Unfortunately, I also see no way to modify the initialization process of those static finals out of the box.
Persisting the initialized state would require you to refactor your code and make all those static objects non static.
